aws-s3-transfer-manager/src/config.rs

```diff
@@ -43,6 +46,11 @@ impl Config {
         &self.concurrency
     }
 
+    /// Returns the name of the app that is using the transfer manager, if it was provided.
+    pub fn app_name(&self) -> Option<&AppName> {
```

Comment:
**fix**: I don't think we want to expose app name. (1) we allow taking an S3 client (today anyway) that can already have app name configured, (2) I don't think this is what mountpoint will be looking for out of a UA header.

Is there something I'm missing here about why we are exposing this?

Reply:
From my reading of the sdk, the app name is just the string for the SDK to add in the user agent with some format check.

1. I think if we take the s3 client and want to respect the setting from transfer manager, we will be adding the user agent setting from transfer manager to the passed in client as the app name.

2. what do you think mountpoint will be looking for? 
I assume they just want a string, and I thought the app name is the wrap of string for SDK to be provided in the user agent.

Reply:
> From my reading of the sdk, the app name is just the string for the SDK to add in the user agent with some format check.

App name (`appId`) comes from the UA SEP. It's meant for end user applications (though I've yet to see someone use it 🤷‍♂️  in any SDK). Mountpoint _could_ go through that but they are more like framework metadata. 


> I think if we take the s3 client and want to respect the setting from transfer manager, we will be adding the user agent setting from transfer manager to the passed in client as the app name.

The issue is the user _could_ have set app name on the client. I think maybe we just leave this off for now and gather requirements from mountpoint to guide us. I _think_ [this](https://github.com/awslabs/mountpoint-s3/blob/993b0d42c0b172fb82fea39ac3964a1b3c74d4cf/mountpoint-s3-client/src/user_agent.rs) is their user agent implementation though. From the looks of it we capture a lot of the md/* fields they are gathering I think so maybe they just need `lib/mountpoint-s3-client#<version>`.

Reply:
the framework metadata is supposed to be used as frameworks **owned** by the SDKs.

> For frameworks owned by the SDKs, framework metadata MUST be included in the header following this format: "lib" / name ["#" version]. Additional metadata MAY be appended.

It seems to be an implementation detail for SDKs to me. And mountpoint is more like the application user, but yeah, they are more like a framework for the end user?
 I guess we can chat with them and see what are they expecting?